### PR TITLE
fix ck build for 5.6.1

### PR DESCRIPTION
**Pull request by afzpatel**
This pr is to fix the following error which occurs when building composable-kernel@5.6.1:
```
make[3]: Entering directory '/tmp/root/spack-stage/spack-stage-composable-kernel-5.6.1-x5elgeyjsssifrim2shnwapog5kxgplh/spack-build-x5elgey'
make[3]: *** No rule to make target 'device_batched_gemm_multi_d_instance', needed by 'CMakeFiles/instances'.  Stop.
make[3]: Leaving directory '/tmp/root/spack-stage/spack-stage-composable-kernel-5.6.1-x5elgeyjsssifrim2shnwapog5kxgplh/spack-build-x5elgey'
make[2]: *** [CMakeFiles/Makefile2:3925: CMakeFiles/instances.dir/all] Error 2
make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-composable-kernel-5.6.1-x5elgeyjsssifrim2shnwapog5kxgplh/spack-build-x5elgey'
make[1]: *** [CMakeFiles/Makefile2:3932: CMakeFiles/instances.dir/rule] Error 2
make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-composable-kernel-5.6.1-x5elgeyjsssifrim2shnwapog5kxgplh/spack-build-x5elgey'
make: *** [Makefile:293: instances] Error 2
```

### CODE REVIEW

**Review by srekolam — APPROVED** [review]

**Comment by afzpatel** [comment]
@spackbot rerun pipeline

**Review by srekolam — APPROVED** [review]

**Comment by afzpatel** [comment]
@spackbot rerun pipeline

**Comment by afzpatel** [comment]
@tldahlgren this pr is ready to be reviewed

File: var/spack/repos/builtin/packages/composable-kernel/package.py [hunk 1]
```diff
@@ -64,9 +64,14 @@ def cmake_args(self):
         ]
         if "auto" not in self.spec.variants["amdgpu_target"]:
             args.append(self.define_from_variant("AMDGPU_TARGETS", "amdgpu_target"))
+        if "@5.6.1:" in self.spec:
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/composable-kernel/package.py]: You should be using `satisfies` here:

```suggestion
        if self.spec.satisfies("@5.6.1:"):
```

File: var/spack/repos/builtin/packages/composable-kernel/package.py [hunk 2]
```diff
@@ -64,9 +64,14 @@ def cmake_args(self):
...
+            args.append(self.define("INSTANCES_ONLY", "ON"))
         return args
 
     def build(self, spec, prefix):
         with working_dir(self.build_directory):
             # only instances is necessary to build and install
-            make("instances")
+            if "@5.6.1:" in self.spec:
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/composable-kernel/package.py]: Same here for use of `satisfies()`.

File: var/spack/repos/builtin/packages/composable-kernel/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/composable-kernel/package.py]: See docs for checking specs at https://spack.readthedocs.io/en/latest/packaging_guide.html#testing-spec-constraints.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Please use `satisfies` for checking spec's version.

**Review by srekolam — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]
LGTM

**Pull request merged**